nwbinspector/utils.py

```diff
@@ -2,18 +2,30 @@
 import os
 import re
 import json
-import numpy as np
-from typing import TypeVar, Optional, List, Dict, Callable
+from typing import TypeVar, Union, Optional, List, Dict, Callable
 from pathlib import Path
 from importlib import import_module
 from packaging import version
 from time import sleep
+from functools import lru_cache
+
+import numpy as np
+from numpy.typing import ArrayLike
+from hdmf.common import VectorData
+
 
 PathType = TypeVar("PathType", str, Path)  # For types that can be either files or folders
 FilePathType = TypeVar("FilePathType", str, Path)
 OptionalListOfStrings = Optional[List[str]]
 
 dict_regex = r"({.+:.+})"
+MAX_CACHE_ITEMS = 1000  # lru_cache default is 128 calls of matching input/output, but might need more to get use here
+
+
+@lru_cache(maxsize=MAX_CACHE_ITEMS)
+def _subsample_data(data: np.ndarray, nelems: Optional[int] = 200):
```

Comment:
```suggestion
def _subsample_data(data: Union[np.ndarray, h5py.Dataset], nelems: Optional[int] = 200):
```

the whole point here is to cache h5py dataset values, right?

Reply:
I also think I'd prefer to take a tuple of slice objects, so we can cache data more generally

Reply:
Done